packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1695,6 +1730,164 @@ export class TransactionController extends BaseController<
       currentNetworkIsEIP1559Compatible && currentAccountIsEIP1559Compatible
     );
   }
+
+  private async updatePostTransactionBalance(
+    transactionMeta: TransactionMeta,
+    { numberOfAttempts = 6 },
+  ) {
+    const transactionId = transactionMeta.id;
+    const postTransactionBalance = await query(this.ethQuery, 'getBalance', [
+      transactionMeta.txParams.from,
+    ]);
+    const latestTransactionMeta = this.getTransaction(
+      transactionId,
+    ) as TransactionMeta;
+    const approvalTransactionMeta = latestTransactionMeta.approvalTxId
+      ? this.getTransaction(latestTransactionMeta.approvalTxId)
+      : null;
+    latestTransactionMeta.postTxBalance = postTransactionBalance.toString(16);
+    const isDefaultTokenAddress = isSwapsDefaultTokenAddress(
+      transactionMeta.destinationTokenAddress as string,
+      transactionMeta.chainId,
+    );
+
+    if (
+      isDefaultTokenAddress &&
+      transactionMeta.preTxBalance === latestTransactionMeta.postTxBalance &&
+      numberOfAttempts > 0
+    ) {
+      setTimeout(() => {
+        // If postTxBalance is the same as preTxBalance, try it again.
+        this.updatePostTransactionBalance(transactionMeta, {
+          numberOfAttempts: numberOfAttempts - 1,
+        });
+      }, UPDATE_POST_TX_BALANCE_TIMEOUT);
+    } else {
+      this.updateTransaction(
+        latestTransactionMeta,
+        'TransactionController#updatePostTransactionBalance - Add post transaction balance',
+      );
+      this.hub.emit(TransactionEvent.postTransactionBalanceUpdated, {
```

Comment:
If we move this method into `utils/swaps.ts`, we could keep this event emit here but in a `then` callback once the method resolves since this is always done at the end.

Reply:
Done https://github.com/MetaMask/core/pull/1877/commits/127444c3b779230c85656050ea48613c3ff3de71